src/components/AddToDo.js

```diff
@@ -0,0 +1,79 @@
+import React, { Fragment } from 'react';
+import { View, Text, TextInput } from 'react-native';
+import Button from './Button';
+import styles from './AddToDo.styles';
+import Header from './Header';
+import strings from '../assets/strings';
+
+class AddToDo extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { taskTitle: '', taskDescription: '', error: '' };
+  }
+
+  validateNewTask = () => {
+    const { addToDo } = this.props;
+    const { emptyTextOrDescription } = strings;
+    if (this.state.taskTitle.length !== 0 && this.state.taskDescription.length !== 0) {
+      addToDo(this.state.taskTitle, this.state.taskDescription);
+    } else {
+      this.setState({ error: emptyTextOrDescription });
+    }
+  }
+
+  renderError = () => {
+    const { errorContainer, errorStyle } = styles;
+    if (this.state.error.length !== 0) {
+      return <View style={errorContainer}><Text style={errorStyle}>{this.state.error}</Text></View>;
+    }
+    return <View />;
+  }
+
+  render() {
+    const { onClickAction } = this.props;
+    const {
+      container, title, description, actions, borderBottom, borderText, rightActionContainer, leftActionContainer, centerContainer,
+    } = styles;
+    const { save, cancel, newTask } = strings;
+    return (
+      <Fragment>
+        <Header>
+          <View style={actions}>
+            <Button onClickAction={onClickAction}>
+              <Text style={leftActionContainer}> {cancel} </Text>
+            </Button>
+          </View>
+          <View style={actions}>
+            <Text style={centerContainer}>{newTask} </Text>
+          </View>
+          <View style={actions}>
+            <Button onClickAction={() => this.validateNewTask()}>
```

Comment:
Here you can simply pass `this.validateNewTask`, taking care to bind it to this component on the constructor. In fact, it's a bad idea to pass arrow functions as props, hinted on [this page](https://reactjs.org/docs/faq-functions.html#arrow-function-in-render). Essentially, each time this `render` function runs, a different `onClickAction` prop is sent to the button, which will probably make it re-render, even if essentially nothing has changed.

Reply:
This is because Solo and Pera think that it would be better do this and avoid to bind every function in the constructor for a better readability. Maybe we need to discuss together what to do with this cases.

Reply:
I agree, we should discuss